### ISSUE TITLE
Timestamp calculation in SenMLResolver class

### ISSUE THREAD

**Issue by JaroslawLegierski** [eclipse/leshan#1301]
Our colleagues identified  issue in timestamp calculation mechanism in Leshan. In class SenMLResolver resolvedTimestamp is calculated as  sum of currentTimestamp (in miliseconds) and resolvedTimestamp (in seconds):

```java
           if (resolvedTimestamp < 268_435_456) {
                resolvedTimestamp = currentTimestamp + resolvedTimestamp;
            }
```

what is your opinion on the following change:

```java
            if (resolvedTimestamp < 268_435_456) {
                resolvedTimestamp =  TimeUnit.MILLISECONDS.toSeconds(currentTimestamp) + resolvedTimestamp;
            }
```
on line 52 ?

**Comment by sbernard31** [comment]
Good catch !

Yep, fix could be something like :  
line 52 : 
```java
resolvedTimestamp =  TimeUnit.MILLISECONDS.toSeconds(currentTimestamp) + resolvedTimestamp;
```
OR line 25 : 
```java
    private long currentTimestamp =  TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
```

**Comment by sbernard31** [comment]
Fixed in `master` by PR #1303 (commit b14a7406f5b12f98c577465605f7e6c20db3f36f)

Thx to you and  your colleagues :wink:
